### PR TITLE
fix the filter-based points remove operation in proxy segment

### PR DESCRIPTION
**Pull request by generall**

### CODE REVIEW

File: lib/collection/src/collection_manager/holders/proxy_segment.rs [hunk 1]
```diff
@@ -473,23 +465,39 @@ impl SegmentEntry for ProxySegment {
...
+        let expected_deleted_count = if total_wrapped_size > 0 {
+            (wrapped_segment_count.exp as f64
+                * (deleted_points_count as f64 / total_wrapped_size as f64)) as usize
+        } else {
+            0
+        };
+
         CardinalityEstimation {
             primary_clauses: vec![],
             min: wrapped_segment_count
```
Comment (IvanPleshkov) [inline lib/collection/src/collection_manager/holders/proxy_segment.rs:494]: Why not like in exp?
```
min: (wrapped_segment_count.min + write_segment_count.min)
                .saturating_sub(deleted_points_count),
```
I guess it is possible to get `exp < min` in current version. But I'm not sure for now

Comment (IvanPleshkov) [reply lib/collection/src/collection_manager/holders/proxy_segment.rs:494]: Oh, I see that `expected_deleted_count` cannot be more than `wrapped_segment_count.exp` - we don't need `saturating_sub ` and my first comment is obsolete

File: lib/segment/src/index/field_index/mod.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ mod stat_tools;
 
 pub use field_index_base::*;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
```
[the remark this answers is not in the document]
Comment (generall) [reply lib/segment/src/index/field_index/mod.rs:17]: forgot to use it

File: lib/collection/src/collection_manager/optimizers/segment_optimizer.rs [hunk 3]
```diff
@@ -413,7 +413,7 @@ pub trait SegmentOptimizer {
             // This block locks all operations with collection. It should be fast
             let mut write_segments_guard = segments.write();
             let deleted_points = proxy_deleted_points.read();
-            let points_diff = already_remove_points.difference(&deleted_points);
+            let points_diff = deleted_points.difference(&already_remove_points);
```
[the remark this answers is not in the document]
Comment (generall) [reply lib/collection/src/collection_manager/optimizers/segment_optimizer.rs:416]: yes, that was a bug

**Review by IvanPleshkov — APPROVED** [review]

File: lib/collection/src/collection_manager/optimizers/segment_optimizer.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (agourlay) [reply lib/collection/src/collection_manager/optimizers/segment_optimizer.rs:416]: that was super sneaky, is it possible to write a tiny unit test here?

**Pull request merged**